### PR TITLE
Added docs and examples about fragment controllers

### PR DESCRIPTION
**Pull request by aschempp**
depends on https://github.com/contao/docs/pull/226 for the caching infos

### CODE REVIEW

File: docs/dev/guides/fragment-controllers.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        parent::__construct($model, $section);
+
+        return new Response($this->generate());
+    }
+    
+    // Do whatever you want here, e.g. override parent methods
+}
+```
+
+{{%expand "Code example to extend a content" %}}
```
Comment (fritzmg) [inline docs/dev/guides/fragment-controllers.md]:
```suggestion
{{%expand "Code example to extend a legacy content element" %}}
```

**Review by leofeyer — DISMISSED** [review]

File: docs/dev/guides/fragment-controllers.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+---
+
+
+## Intro
+
+A regular page in Contao consists of various partials that build the actual page content:
+
+1. The page has a layout assigned, which renders the base template (usually `fe_page.html5`)
+2. Each page layout has multiple layout sections (header, left, main, custom ones etc.)
+3. Each layout sections contain one or multiple modules or articles
```
Comment (Toflar) [inline docs/dev/guides/fragment-controllers.md]: Each layout section contains

**Review by fritzmg — CHANGES_REQUESTED** [review]

File: docs/dev/guides/fragment-controllers.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+## Concept
+
+In Symfony, an HTML template is usually generated by a controller. From the Symfony documentation:
+
+> A controller is a PHP function you create that reads information from the Request object and
+> creates and returns a Response object. The response could be an HTML page, JSON, XML, a
+> file download, a redirect, a 404 error or anything else. The controller executes whatever
+> arbitrary logic your application needs to render the content of a page.
+
+A normal controller has a route – a URL on your site – that will generate the response.
```
Comment (Toflar) [inline docs/dev/guides/fragment-controllers.md]:
```suggestion
A normal controller has a route – an URL on your site – that will generate the response.
```

File: docs/dev/guides/fragment-controllers.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+
+## Fragment Controllers
+
+Contao provides two different types of fragments out-of-the-box:
+[Front End Modules][moduels] and [Content Elements][elements]. For each of
+these, there are abstract classes (namely `AbstractFrontendModuleController`
+and `AbstractContentElementController`) that mimick the behavior of a Contao 3
+module/element. They generate the Contao template based on the request and current
+model, and allow to
```
Comment (Toflar) [inline docs/dev/guides/fragment-controllers.md]: allow to what?

File: docs/dev/guides/fragment-controllers.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+
+
+## Fragment Controllers
+
+Contao provides two different types of fragments out-of-the-box:
+[Front End Modules][moduels] and [Content Elements][elements]. For each of
+these, there are abstract classes (namely `AbstractFrontendModuleController`
+and `AbstractContentElementController`) that mimick the behavior of a Contao 3
+module/element. They generate the Contao template based on the request and current
```
Comment (Toflar) [inline docs/dev/guides/fragment-controllers.md]: I don’t think mentioning v3 is relevant

File: docs/dev/guides/fragment-controllers.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+namespace App\Controller\ContentElement;
+
+use Contao\ContentGallery;
+use Contao\ContentModel;
+use Contao\CoreBundle\ServiceAnnotation\FrontendModule;
+use Symfony\Component\HttpFoundation\Response;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+/**
+ * @FrontendModule(category="news")
```
Comment (richardhj) [inline docs/dev/guides/fragment-controllers.md]: has to be CE annotation here

File: docs/dev/guides/fragment-controllers.md (hunk 5 above)
Comment (aschempp) [reply docs/dev/guides/fragment-controllers.md]: what else would you call it? *old-school Contao module/element*? Or legacy?

File: docs/dev/guides/fragment-controllers.md (hunk 6 above)
Comment (aschempp) [reply docs/dev/guides/fragment-controllers.md]: fixed in 990eff98c449113f133e2811d7a406aeb7fa01f7

File: docs/dev/guides/fragment-controllers.md (hunk 5 above)
Comment (aschempp) [reply docs/dev/guides/fragment-controllers.md]: nevermind, updated in 0f29eab5055d1f6ea24c00ac6ab144527302da84 :)

**Comment by aschempp** [comment]
Not sure how I can revert the changes to the hugo theme @fritzmg ?

**Review by fritzmg — APPROVED** [review]

**Comment by fritzmg** [comment]
Updated Learn theme should be fine, thank you @aschempp and @Toflar :)

**Pull request merged**